sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -1373,10 +1375,22 @@ public void processElement(ProcessContext c) {
   @VisibleForTesting
   static class WriteToSpannerFn extends DoFn<Iterable<MutationGroup>, Void> {
 
-    private transient SpannerAccessor spannerAccessor;
     private final SpannerConfig spannerConfig;
     private final FailureMode failureMode;
 
+    // Only create one SpannerAccessor for each different SpannerConfig.
```

Comment:
Setup for SpannerAccessor are done from BatchSpannerRead, CreateTransactionFn, NaiveSpannerRead, ReadSpannerSchema.

Should this go into SpannerAccessor directly and when create() is called, should that take care of creating if absent rather than SpannerIO? let me know your thoughts.

Reply:
Good call -- moved to SpannerAccessor. 
To make it clearer, SpannerAccessor.create(SpannerConfig) has been renamed to getOrCreate(SpannerConfig).